eth/beacon/helpers.py

```diff
@@ -20,3 +170,100 @@ def get_active_validator_indices(dynasty: int,
         if (validator.start_dynasty <= dynasty and dynasty < validator.end_dynasty):
             o.append(index)
     return o
+
+
+#
+# Shuffling
+#
+def shuffle(lst: List[Any],
+            seed: Hash32) -> List[Any]:
+    lst_count = len(lst)
+    assert lst_count <= 16777216
+    o = [x for x in lst]
+    source = seed
+    i = 0
+    while i < lst_count:
+        source = blake(source)
+        for pos in range(0, 30, 3):
+            m = int.from_bytes(source[pos:pos + 3], 'big')
+            remaining = lst_count - i
+            if remaining == 0:
+                break
+            rand_max = 16777216 - 16777216 % remaining
+            if m < rand_max:
+                replacement_pos = (m % remaining) + i
+                o[i], o[replacement_pos] = o[replacement_pos], o[i]
+                i += 1
+    return o
+
+
+@to_tuple
+def split(lst: List[Any], number: int) -> List[Any]:
```

Comment:
I believe this function is the same as `cytoolz.partition_all`

Reply:
Ah, I still think you can do this in a cleaner way with `partition_all` (though I did initially misunderstand what this function was doing).

```python
def split(seq, chunks):
    chunk_size = len(seq) // chunks
    if chunk_size < 1:
        raise ValueError("Cannot split into chunks, not enough values")
    return partition_all(chunk_size, seq)
```

Reply:
Oh, and the `Any` should be updated to use a generic type.

```python
from typing import TypeVar
TItem = TypeVar('TItem')
def split(seq: Sequence[TItem], number: int) -> Iterable[TItem]:
    ...
```

This allows us to maintain type safety for values that pass through this function.  The `shuffle` function above should be updated in the same manner (and can use the same generic type).

Reply:
I forgot to mention that `split` function is [in-protocol](https://github.com/ethereum/eth2.0-specs/blob/master/specs/casper_sharding_v2.1.md), so we still have to produce the *same* result as the old split function after refactorings. Note that with `split()`, the end chunks have more elements than the front chunks. I think stick with the spec is safer.